src/main/java/net/atos/zac/app/zaken/ZakenRESTService.java

```diff
@@ -644,18 +658,41 @@ public RESTBesluit createBesluit(final RESTBesluitVastleggenGegevens besluitToev
     @Path("besluit")
     public RESTBesluit updateBesluit(final RESTBesluitWijzigenGegevens restBesluitWijzgenGegevens) {
         final Zaak zaak = zrcClientService.readZaak(restBesluitWijzgenGegevens.zaakUuid);
+        // assertActie(policyService.readZaakActies(zaak).getBesluitWijzigen()); //todo policy aanmaken
         final Besluit besluit = brcClientService.readBesluit(restBesluitWijzgenGegevens.besluitUuid);
-        if (!besluit.getZaak().equals(zaak.getUrl())) {
-            throw new IllegalStateException("Zaak en besluit horen niet bijelkaar");
-        }
         besluit.setToelichting(restBesluitWijzgenGegevens.toelichting);
-        besluit.setVervaldatum(restBesluitWijzgenGegevens.vervaldatum);
         besluit.setIngangsdatum(restBesluitWijzgenGegevens.ingangsdatum);
+        besluit.setVervaldatum(restBesluitWijzgenGegevens.vervaldatum);
         Besluit updatedBesluit = brcClientService.updateBesluit(besluit);
+        if (zaak.getResultaat() != null) {
+            final Resultaat zaakResultaat = zrcClientService.readResultaat(zaak.getResultaat());
+            final Resultaattype resultaattype = ztcClientService.readResultaattype(restBesluitWijzgenGegevens.resultaattypeUuid);
+            if (zaakResultaat.getResultaattype() != resultaattype.getUrl()) {
+                zrcClientService.deleteResultaat(zaakResultaat.getUuid());
+                zgwApiService.createResultaatForZaak(zaak, restBesluitWijzgenGegevens.resultaattypeUuid, StringUtils.EMPTY);
```

Comment:
Is het niet beter om null als toelichting mee te geven? Dan geef je expliciet aan dat je geen toelichting mee geeft. Nu geef je een lege toelichting mee.